apps/explorer/src/components/Activity/TransactionsActivityTable.tsx

```diff
@@ -0,0 +1,133 @@
+        () => rpc.getTotalTransactionBlocks(),
+        { cacheTime: 24 * 60 * 60 * 1000, staleTime: Infinity, retry: false }
+    );
+
+    const cardData =
+        data && Boolean(data.pages[currentPage])
+            ? genTableDataFromTxData(data.pages[currentPage].data)
+            : undefined;
+
+    return (
+        <div data-testid="tx">
+            <div className="flex flex-col space-y-5 text-left xl:pr-10">
+                {isLoading || isFetching || !cardData ? (
+                    <PlaceholderTable
+                        rowCount={limit}
+                        rowHeight="16px"
+                        colHeadings={[
+                            'Digest',
+                            'Sender',
+                            'Txns',
+                            'Gas',
+                            'Time',
+                        ]}
+                        colWidths={['30%', '30%', '10%', '20%', '10%']}
+                    />
+                ) : (
+                    <div>
+                        <TableCard
+                            data={cardData.data}
+                            columns={cardData.columns}
+                        />
+                    </div>
+                )}
+
+                <div className="flex justify-between">
+                    {(hasNextPage || data?.pages.length) &&
+                    !disablePagination ? (
+                        <Pagination
+                            onNext={() => {
+                                if (isLoading || isFetching) {
+                                    return;
+                                }
+
+                                // Make sure we are at the end before fetching another page
+                                if (
+                                    data &&
+                                    currentPage === data?.pages.length - 1
+                                ) {
+                                    fetchNextPage();
+                                }
+                                setCurrentPage(currentPage + 1);
+                            }}
+                            hasNext={
+                                Boolean(hasNextPage) &&
+                                Boolean(data?.pages[currentPage])
+                            }
+                            hasPrev={currentPage !== 0}
+                            onPrev={() => setCurrentPage(currentPage - 1)}
+                            onFirst={() => setCurrentPage(0)}
+                        />
+                    ) : (
+                        disablePagination && (
+                            <Link to="/recent" after={<ArrowRight12 />}>
+                                More Transaction Blocks
+                            </Link>
+                        )
+                    )}
+
+                    <div className="flex items-center space-x-3">
+                        <Text variant="body/medium" color="steel-dark">
+                            {countQuery.data
```

Comment:
small nit here, probably can move into its own variable so it looks a bit cleaner